frontend/src/routes/Governance/discovered/grouping.ts

```diff
@@ -0,0 +1,205 @@
+  }
+
+  const getPolicySource = (policy: any, helmReleases: any[], channels: any[], subscriptions: any[]): ISourceType => {
+    const parentPolicy = getStringLabelValue(policy.label, 'policy.open-cluster-management.io/policy')
+    const parentNs = parentPolicy?.split('.')[0] ?? ''
+    const parentName = parentPolicy?.split('.')[1] ?? ''
+
+    if (parentNs && parentName) {
+      return {
+        type: 'Policy',
+        parentNs,
+        parentName,
+      }
+    }
+
+    const isExternal = policy._isExternal
+    let source: string | JSX.Element = 'Local'
+    if (isExternal) {
+      const policySource = resolveSource(policy.annotations ?? {}, helmReleases, channels, subscriptions)
+      source = policySource ? getSourceText(policySource, isExternal) : 'Managed externally'
+    }
+    // source is string
+    return {
+      type: source,
+      parentNs: '',
+      parentName: '',
+    }
+  }
+
+  enum PolicySeverity {
+    Unknown,
+    Low,
+    Medium,
+    High,
+    Critical,
+  }
+
+  const severityTable = (severity: string): number => {
+    if (!severity) return PolicySeverity.High
+
+    switch (severity.toLocaleLowerCase()) {
+      case 'low':
+        return PolicySeverity.Low
+      case 'medium':
+        return PolicySeverity.Medium
+      case 'high':
+        return PolicySeverity.High
+      case 'critical':
+        return PolicySeverity.Critical
+      default:
+        // unknown or null
+        return PolicySeverity.High
+    }
+  }
+
+  const createMessage = (data: any[], helmReleases: any[], channels: any[], subscriptions: any[]): any[] => {
+    const policiesWithSource = JSON.parse(JSON.stringify(data), (_, v: string) => {
+      if (v === 'true') return true
+      return v === 'false' ? false : v
+    }).map((policy: any): any => {
+      return {
+        ...policy,
+        remediationAction: policy.remediationAction || 'inform', // Default is inform
+        source: getPolicySource(policy, helmReleases, channels, subscriptions),
+      }
+    })
+
+    const groupByNameKind: { [nameKind: string]: any[] } = {}
+    // Group by clusterName and kind
+    policiesWithSource.forEach((p: any) => {
+      const nameKind: string = p['name'] + p['kind'] || ''
+      const existingGroup = groupByNameKind[nameKind] || []
+      existingGroup.push(p)
+      groupByNameKind[nameKind] = existingGroup
+    })
+
+    const keys = Object.keys(groupByNameKind)
+
+    return keys.map((nameKind) => {
+      const group = groupByNameKind[nameKind] || []
+
+      let highestSeverity = 0
+      let responseAction = group[0].remediationAction
+      let source = { ...group[0].source }
+
+      for (const policy of group) {
+        // Set to highest severity among grouped policies
+        if (severityTable(policy.severity) > highestSeverity) {
+          highestSeverity = severityTable(policy.severity)
+        }
+
+        if (policy.remediationAction != responseAction) {
+          responseAction = 'enforce/inform'
```

Comment:
I think this is how it's presented in the existing policies table.
```suggestion
          responseAction = 'inform/enforce'
```